File: igaServer/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace igaServer.Models
{
    public class Product
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        public string ImageUrl { get; set; }

        public string Category { get; set; } // "Veggie", "Meat"

        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; } // 单价/卖价

        /// <summary>成本价（仅后台可见，用于计算利润率）</summary>
        [Column(TypeName = "decimal(18,2)")]
        public decimal CostPrice { get; set; } // 成本价

        public string Unit { get; set; } // kg, ea, box

        public int StockQuantity { get; set; } // 库存

        public bool IsActive { get; set; } = true; // 上下架

        /// <summary>Stripe Product ID（后台同步后写入）</summary>
        [StringLength(64)]
        public string? StripeProductId { get; set; }

        /// <summary>Stripe Price ID（结账行项目用）</summary>
        [StringLength(64)]
        public string? StripePriceId { get; set; }

        // --- 核心业务字段 ---
        public bool IsWeighingRequired { get; set; } // 是否是称重商品（决定是否需要二次退款）
    }
}

File: frontEnd/src/types/product.ts
export interface ProductUnitPriceOption {
  unit: string;
  price: number;
}

/** 商品类型，与后端 Product 模型对应（后台含 costPrice） */
export interface Product {
  id: number;
  name: string;
  imageUrl?: string;
  category: string;
  price: number;
  costPrice?: number; // 成本价，仅后台可见
  unit: string;
  unitPriceOptionsJson?: string;
  unitPriceOptions?: ProductUnitPriceOption[];
  isActive: boolean;
  isWeighingRequired?: boolean;
  /** 称重商品默认预估重量（kg），用于顾客端展示与加购初值 */
  defaultExpectedWeightKg?: number;
  /** 商品描述，显示在商品卡上 */
  description?: string;
}

/** 商品表单值（新增/编辑） */
export interface ProductFormValues {
  name: string;
  imageUrl: string;
  category: string;
  price: number;
  costPrice?: number;
  unit: string;
  unitPriceOptions: ProductUnitPriceOption[];
  unitPriceOptionsJson?: string;
  isActive: boolean;
  isWeighingRequired: boolean;
  defaultExpectedWeightKg?: number;
  description?: string;
}

/** 分页响应 */
export interface PaginatedResponse<T> {
  items: T[];
  total: number;
  page: number;
  pageSize: number;
}
